### DISCUSSION TITLE
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by jdepons** [GMOD/jbrowse-components#3886] [Q&A]
I am loading large variant tracks into JBrowse 2.   When I get to the indexing step I get a fatal error.

**FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory**

Is there a way I can tell the JBrowse CLI to use more RAM when indexing?  As an example,  in the Java world I can use the  -Xmx option to tell the program to allocate more RAM.  I am wondering if there is a similar option I can pass the JBrowse 2 CLI.

**Comment by garrettjstevens** [comment]
Yes, since the JBrowse CLI uses Node.js, you can use the `NODE_OPTIONS` environment variable to use more RAM. For example:

```sh
NODE_OPTIONS='--max-old-space-size=4096' jbrowse text-index
```

where `4096` is the size in MiB. The default size varies by Node.js version and machine, but I think it's often somewhere in the 512–2048 MiB range, so you'll want to go higher than that.

**Reply by cmdcolin** [reply]
I believe it is --max-old-space-size (dashes instead of underscores) xref https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-megabytes

**Reply by garrettjstevens** [reply]
Oops, thanks, fixed now.

**Comment by cmdcolin** [comment]
@garrettjstevens  provides the right approach for changing memory allocation, but there could have been a bug or memory leak. I just released a new version of our ixixx package that may help with this issue...if you re-run `npm install -g @jbrowse/cli` it should pick up the new latest version, and it may help!
